velox/common/memory/SharedArbitrator.h

```diff
@@ -282,12 +292,29 @@ class SharedArbitrator : public memory::MemoryArbitrator {
       uint64_t& maxGrowTarget,
       uint64_t& minGrowTarget);
 
-  // Invoked to get or refresh the memory stats of the candidate memory pools
-  // for arbitration. If 'freeCapacityOnly' is true, then we only get free
-  // capacity stats for each candidate memory pool.
-  void getCandidateStats(
-      ArbitrationOperation* op,
-      bool freeCapacityOnly = false);
+  // Invoked to get or refresh the candidate memory pools for arbitration. If
+  // 'freeCapacityOnly' is true, then we only get free capacity stats for each
+  // candidate memory pool.
+  void getCandidates(ArbitrationOperation* op, bool freeCapacityOnly = false);
+
+  // Sorts 'candidates' based on reclaimable free capacity in descending order.
+  static void sortCandidatesByReclaimableFreeCapacity(
+      std::vector<Candidate>& candidates);
+
+  // Sorts 'candidates' based on reclaimable used capacity in descending order.
+  static void sortCandidatesByReclaimableUsedCapacity(
+      std::vector<Candidate>& candidates);
+
+  // Sorts 'candidates' based on actual used memory in descending order.
+  static void sortCandidatesByUsage(std::vector<Candidate>& candidates);
+
+  // Finds the candidate with the largest capacity. For 'requestor', the
+  // capacity for comparison including its current capacity and the capacity to
+  // grow.
+  static const SharedArbitrator::Candidate& findCandidateWithLargestCapacity(
+      MemoryPool* requestor,
+      uint64_t targetBytes,
+      const std::vector<Candidate>& candidates);
```

Comment:
Why do we change them to static? Is it possible to keep them as anonymous namespace helper in .cpp?

Reply:
I tried to keep Candidate private and used friend class annotation for test case. But it didn't work somehow. But will try later. It is fine to keep them as static method as well.